tensorflow_model_optimization/python/core/sparsity_tf2/sparse_utils_test.py

```diff
@@ -0,0 +1,223 @@
+    self.assertAllClose(expected_initialization, average_initialization, rtol=204e-3)
+
+  @parameterized.parameters(
+    zip(itertools.product((0.4,), NOSMALL_SHAPES), MIDPOINTS)
+  )
+  def testMidpointInitialOnesPermuteOnes(self, ratio_shape_pair, midpoint):
+    # Given all indices that are masked according to ratio,
+    # the average value of all indices should appear at the "center"
+    # of the given matrix. Similarly to above test, the ratio
+    # of 1's in the mask is not deciding factor.
+    ratio, shape = ratio_shape_pair
+    matrix = tf.ones(shape)
+    self.assertAllEqual(shape, matrix.shape)
+
+    seed = 0
+    sparse_matrices_perm = []
+    for _ in range(400):
+      matrix_sparse_permute = sparse_utils.PermuteOnes(ratio)(shape, seed=seed)
+      matrix_midpoint_idx = tf.where(tf.reshape(matrix_sparse_permute, (-1,)) == 1)
+      sparse_matrices_perm.append(tf.math.reduce_mean(matrix_midpoint_idx))
+      seed += 1
+    average_midpoints = tf.math.reduce_mean(sparse_matrices_perm, axis=0)
+    expected_midpoints = tf.constant(midpoint) # broadcasts
+    # expect that each entry in the matrix is "on"/1 in expectation, ratio number of times
+    self.assertAllClose(expected_midpoints, average_midpoints, atol=1.)
+
+  @parameterized.parameters(
+    zip(itertools.product((0.4,), NOSMALL_SHAPES), MIDPOINTS)
+  )
+  def testMidpointInitialOnesBernouilli(self, ratio_shape_pair, midpoint):
+    ratio, shape = ratio_shape_pair
+    matrix = tf.ones(shape)
+    self.assertAllEqual(shape, matrix.shape)
+
+    seed = 0
+    sparse_matrices_bern = []
+    for _ in range(400):
+      matrix_sparse_bernouilli = sparse_utils.Bernouilli(ratio)(shape, seed=seed)
+      matrix_midpoint_idx = tf.where(tf.reshape(matrix_sparse_bernouilli, (-1,)) == 1)
+      sparse_matrices_bern.append(tf.math.reduce_mean(matrix_midpoint_idx))
+      seed += 1
+    average_midpoint = tf.math.reduce_mean(sparse_matrices_bern, axis=0)
+    expected_midpoint = tf.constant(midpoint)
+    self.assertAllClose(expected_midpoint, average_midpoint, atol=1.)
+  
+
+  @parameterized.parameters(
+    ((30, 4), 0.5, 60), 
+    ((1, 2, 1, 4), 0.8, 7), 
+    ((30,), 0.1, 3)
+  )
+  def testBernouilliMatrixFraction(self, shape, ratio, expected_ones):
+    # Check that the mean of ones_count is approximately
+    # ratio * num_elements_in_shape.
+    matrix = tf.ones(shape)
+    self.assertAllEqual(shape, matrix.shape)
+
+    seed = 0
+
+    counts = []
+    for _ in range(400):
+      output = sparse_utils.Bernouilli(ratio)(shape, seed=seed)
+      ones_count = tf.math.count_nonzero(output, dtype=tf.int32)
+      counts.append(ones_count)
+      self.assertAllEqual(output.shape, shape)
+      seed += 1
+    mean_counts = np.mean(counts, axis=0)
+    stdev_counts = np.std(counts, axis=0)
+    self.assertAllClose(mean_counts, expected_ones, rtol=8e-2)
+
+  @parameterized.parameters(
+    ((30, 4), 0.5, 60), 
+    ((1, 2, 1, 4), 0.8, 7), 
```

Comment:
Leave a comment here saying that the expected is `7` because we do `ceil` instead of round. Also document that behavior in the initializer.